src/message_receiver.pyx

```diff
@@ -84,9 +84,14 @@ cdef class cMessageReceiver(StructBase):
             raise RuntimeError("Unable to send message dispostition 'released' for message number {}".format(message_number))
         c_amqpvalue.amqpvalue_destroy(delivery_state)
 
-    cpdef settle_rejected_message(self, c_amqp_definitions.delivery_number message_number, const char* error_condition, const char* error_description):
+    cpdef settle_rejected_message(self, c_amqp_definitions.delivery_number message_number, const char* error_condition, const char* error_description, AMQPValue error_info=None):
         cdef c_amqpvalue.AMQP_VALUE delivery_state
-        delivery_state = c_message.messaging_delivery_rejected(error_condition, error_description)
+        cdef c_amqp_definitions.fields delivery_fields
+        if error_info is not None:
+            delivery_fields = <c_amqp_definitions.fields>error_info._c_value
```

Comment:
It looks like we have two separate code paths for rejecting a message?
Is one of them on callback?

Reply:
AFAIK, we only have `settle_rejected_message` for rejecting..
I don't see a separate code path.

If you're referring to `delivery_rejected` in `message.pyx`, we don't use it in python code. it's for constructing a delivery  reject object.